src/Compilers/VisualBasic/Portable/Symbols/MethodSignatureComparer.vb

```diff
@@ -425,14 +442,23 @@ Done:
             type1 = SubstituteType(typeSubstitution1, type1)
             type2 = SubstituteType(typeSubstitution2, type2)
 
-            If Not type1.Type.IsSameTypeIgnoringCustomModifiers(type2.Type) Then
+            If Not type1.Type.IsSameType(type2.Type, TypeCompareKind.AllIgnoreOptionsForVB) Then
                 Return SymbolComparisonResults.ReturnTypeMismatch
-            ElseIf (comparisons And SymbolComparisonResults.CustomModifierMismatch) <> 0 AndAlso
-                   (type1 <> type2) Then
-                Return SymbolComparisonResults.CustomModifierMismatch
             End If
 
-            Return Nothing
+            Dim result As SymbolComparisonResults = 0
+
+            If (comparisons And SymbolComparisonResults.TupleNamesMismatch) <> 0 AndAlso
+                    Not type1.Type.IsSameType(type2.Type, TypeCompareKind.AllIgnoreOptionsForVB And Not TypeCompareKind.IgnoreTupleNames) Then
+                result = result Or SymbolComparisonResults.TupleNamesMismatch
```

Comment:
I resolved per comment above. `IsSameType` does not take a `stopIfAny` parameter.
If that is not what you meant, please clarify.

---
In reply to: [82649678](https://github.com/dotnet/roslyn/pull/14229#discussion_r82649678) [](ancestors = 82649678,82479037,82478850,82462854)